hazelcast/src/main/java/com/hazelcast/map/AbstractEntryProcessor.java

```diff
@@ -71,4 +75,16 @@ public void processBackup(Map.Entry<K, V> entry) {
             process(entry);
         }
     }
+
+    private class HazelcastInstanceAwareEntryBackupProcessorImpl extends EntryBackupProcessorImpl
+            implements HazelcastInstanceAware {
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
+            final AbstractEntryProcessor<K, V> outer = AbstractEntryProcessor.this;
+            if (outer instanceof HazelcastInstanceAware) {
```

Comment:
The comment is because of a previous version where `AbstractEntryProcessor` implemented `HazelcastInstanceAware` so this check was not needed. Later we figured out that we can't change the type hierarchy of `AbstractEntryProcessor` nor set it's `serialVersionUID`.

Now the check is mandatory since the outer object no longer necessarily implements `HazelcastInstanceAware`.